src/test/ui/closures/2229_closure_analysis/issue-87987.rs

```diff
@@ -0,0 +1,30 @@
+// run-pass
+// edition:2021
+
+struct Props {
+    field_1: u32, //~ WARNING: field is never read: `field_1`
+    field_2: u32, //~ WARNING: field is never read: `field_2`
+}
+
+fn main() {
+    // Test 1
+    let props_2 = Props { //~ WARNING: unused variable: `props_2`
```

Comment:
This warning is inconsistent with what happens when `let _: Props = props_2;` is used outside of a closure. This is a different problem so I have created a new issue for it.

https://github.com/rust-lang/project-rfc-2229/issues/57